dc/s2n-quic-dc/src/path/secret/map/state.rs

```diff
@@ -389,20 +507,41 @@ where
         let id = *entry.id();
         let peer = entry.peer();
 
-        let (same, other) = self.ids.insert(id, entry.clone());
+        // This is the only place that inserts into the ID list.
+        let same = self.ids.insert(entry.clone());
         if same.is_some() {
             // FIXME: Make insertion fallible and fail handshakes instead?
             panic!("inserting a path secret ID twice");
         }
 
-        if let Some(evicted) = other {
-            self.subscriber().on_path_secret_map_id_entry_evicted(
-                event::builder::PathSecretMapIdEntryEvicted {
-                    peer_address: SocketAddress::from(*evicted.1.peer()).into_event(),
-                    credential_id: evicted.1.id().into_event(),
-                    age: evicted.1.age(),
-                },
-            );
+        {
+            let mut queue = self
+                .eviction_queue
+                .lock()
+                .unwrap_or_else(|e| e.into_inner());
+
+            queue.push_back(Arc::downgrade(&entry));
+
+            // We are out of room, need to prune some entries.
+            if queue.len() == self.max_capacity {
```

Comment:
Does this need to be checked before we push? Otherwise `len` will always be `max_capacity - 1`?